stack/src/kernel/edrv/edrv-i210.c

```
@@ -1057,74 +1056,80 @@ tOplkError edrv_getMacTime(UINT64* pCurtime_p)
 
 //------------------------------------------------------------------------------
 /**
-\brief  Set the cycle frequency
-
-The function configure the cycle frequency in the I210 timer.
-
-\param  frequency       The cycle frequency to store into the timer.
-*/
-//------------------------------------------------------------------------------
-void edrv_setCyclicFrequency(UINT32 frequency)
-{
-
-    if (!edrvInstance_l.fInitialized)
-        return;
-
-    EDRV_REGDW_WRITE(EDRV_FREQOUT0, frequency);
-}
-
-//------------------------------------------------------------------------------
-/**
 \brief  Start timer
 
 The function starts the timer in the I210.
 
+\param  index_p         Index of the timer.
 \param  pTimerHdl_p     Timer handle of the timer to start.
 \param  frequency_p     Cycle time (frequency).
 
 \return The function returns a tOplkError error code.
 */
 //------------------------------------------------------------------------------
-tOplkError edrv_startTimer(tTimerHdl* pTimerHdl_p, UINT32 frequency_p)
+tOplkError edrv_startTimer(UINT32 index_p, tTimerHdl* pTimerHdl_p, UINT64 frequency_p)
 {
     UINT32              reg;
     struct timespec     ts;
+    struct timespec*    pTimeout = NULL;
+    UINT64              seconds, nanoseconds, rem;
 
     if (!edrvInstance_l.fInitialized)
         return kErrorOk;
 
-    if (pTimerHdl_p == NULL)
+    if ((pTimerHdl_p == NULL) || (index_p >= EDRV_MAX_TIMER_COUNT))
     {
-        printk("%s() invalid parameter for timer handle\n", __FUNCTION__);
-        return kErrorTimerNoTimerCreated;
+        printk("%s() Invalid parameters for the timer\n", __FUNCTION__);
+        return kErrorTimerInvalidHandle;
     }
 
-    edrvInstance_l.timerHdl = *pTimerHdl_p;
+    pTimeout = &edrvInstance_l.aTimeOut[index_p];
```

Sorry, I meant line 1088. I guess if `edrvInstance_l.aTimerHdl[index_p]` is not 0 we should raise an error.